### PR TITLE
Add AggregateIcaModel class to wrap aggregated models

### PR DESCRIPTION
**Pull request by johnstill**
Adds a class to provide functionality for aggregated ICA Phenotype models, i.e. a model where the phenotypes are pulled from multiple sub-models trained on samples of the same underlying dataset.

### CODE REVIEW

File: cml_data_tools/models.py [hunk 1]
```diff
@@ -7,6 +9,99 @@
 from sklearn.base import BaseEstimator, TransformerMixin
 
 
+def clusters_to_components(clusters, n=2000):
```
Comment (tlasko) [inline cml_data_tools/models.py:12]: Please add docstring. Looks like cml_data_tools/clustering.py needs documentation as well.

File: cml_data_tools/models.py [hunk 2]
```diff
@@ -7,6 +9,99 @@
...
+        for i, (model, ids) in enumerate(zip(models, self.components)):
+            cols = np.array(model.phenotype_names)[ids]
+            ph = model.phenotypes_[cols]
+            ph.columns = self.ids_to_names(i, ids)
+            parts.append(ph)
+        self.phenotypes = pd.concat(parts, axis=1)
+
+    def ids_to_names(self, m, ids):
+        """
+        Produces an array of appropriate identifiers for the phenotypes
```
Comment (tlasko) [inline cml_data_tools/models.py:81]: Specify the what the renaming is in the docstring.

File: cml_data_tools/models.py [hunk 3]
```diff
@@ -7,6 +9,99 @@
...
+
+        Arguments
+        ---------
+        X : pandas.DataFrame
+            Has one row per instance, one column per variable
+
+        Returns
+        -------
+        pandas.DataFrame
+            A pandas dataframe with rows corresponding to (and indexed the same
```
Comment (tlasko) [inline cml_data_tools/models.py:110]: Specify how they are renamed. Fine to just point to the renaming function.

File: cml_data_tools/models.py [hunk 4]
```diff
@@ -7,6 +9,99 @@
...
+            columns = np.array(model.phenotype_names)[ids]
+            x_hat = model.transform(X)[columns]
+            x_hat.columns = self.ids_to_names(i, ids)
+            parts.append(x_hat)
+        return pd.concat(parts, axis=1)
+
+    class ModelCore:
+        """Extracts core functionality from IcaPhenotypeModel"""
+        def __init__(self, model):
+            self.ica = model.ica
```
Comment (tlasko) [inline cml_data_tools/models.py:129]: Do the reference semantics work here? Meaning, does `model` need to be kept around for this to work?

**Review by tlasko — COMMENTED** [review]
Looks fine, with just a few comments. Mostly documentation.

File: cml_data_tools/models.py (hunk 4 above)
Comment (johnstill) [reply cml_data_tools/models.py:129]: Yes, this copies the attributes from `model` to the `ModelCore` instance, which allows us to save this separately without needing to keep the entire `model` instance around.

**Review by tlasko — APPROVED** [review]
LGTM

**Pull request merged**